### ISSUE TITLE
BTT SKR 1.4 Turbo + TMC2209 + BLTouch - homing issues

### ISSUE THREAD

**Issue by ABLomas** [MarlinFirmware/Marlin#18445]
Setup - CR20Pro + SKR1.4 Turbo + TMC2209 + BLTouch
Got it working with 2020.05.01 bugfix-2.0.x branch - no pins cut on TMC drivers, fully sensorless homing (mechanical switches not connected), all fine.
Tried to test new features week ago (06.22 bugfix branch) - could not get sensorless homing to work at all, it simply won't trigger. Decided to test another run now - sensorless homing is kinda working on XY axis (below), but not on Z-axis.

So for now:
 - configs: [configuration.zip](https://github.com/MarlinFirmware/Marlin/files/4840606/configuration.zip)
 - setup was merged from earlier config using winmerge, taking only variables i changed
 - on startup printer decides, that it's "home point" is current location and refuses to go until min endstop (in other words, if i power on printer with it's X and Y at home - all fine, i can move it all way up to max/back to min), but if i power on with axis in middle - it's "minimum point" is at this location and "max" is somewhere beyond printer limits. It kinda stops on max after some effort so i think stallguard is working, but coordinate system is completely off.
If i try to home X and Y axis - it just moves very slightly and decides, that it's "already home".
 - Now bltouch probe is somewhat not working at all. On the same todays build it working OK, but not reporting status properly, so Z-axis did not stop and tried to drill buildplate. Got advice on another bug to cut DIAG pin, did that on Z-axis, now probe is not working anymore from printer side:
I can send comands and probe responds correctly (M280 P0 S120, M280 P0 S90, M280 P0 S10), but does not matter in which state probe is, M119 command reports the same:
```
Send: M119
Recv: Reporting endstop status
Recv: x_min: open
Recv: y_min: open
Recv: z_min: TRIGGERED
Recv: ok
```
This output does not depends on probe position **and** does not has probe status included. Before cutting DIAG pin on Z-axis driver i could home Z-axis, now it just  tests probe several times and stops here (with message "STOP called because of BLTouch error - restart with M999"). Z-axis itself moves OK if i command movement from LCD menu.

So, in short:
 - the same HW configuration was working before on earlier build (SKR1.4 turbo + TMC2209 + sensorless + BLTouch)
 - with new build i could not get homing working at all
 - and made things worse by cutting DIAG pin on TMC2209 (AFAIK every build which uses sensorless homing is not cutting any pins, they work fine), but i will resolder it later

It seems that i actually hit two issues here - one is with "starting point is home point" for XY axis, another - that bltouch does not work anymore with newer code (sort of). Second one may be due to changed board definitions. First one - maybe stallguard current is completely different after library upgrade and it triggers even on light movement?

**Comment by ABLomas** [comment]
OK, so axis not homing is def. sensitivity issue, which changed.
Earlier i had something like 95 on both axis and it was working fine, now i needed to lower them:
```
Send: M914
Recv: X homing sensitivity: 65
Recv: Y homing sensitivity: 57
```

Now X and Y axis home correctly. Big change ;-)
Why Z axis and probe is not working as earlier still not known.

**Comment by ABLomas** [comment]
I tried to clear older pin definitions and yes, they probably changed and this is why probe wasn't visible.
So, now i have:
```
// #define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN
// #define Z_STOP_PIN P0_10
#define USE_PROBE_FOR_Z_HOMING

```
...and M119 output:
```
Recv: Reporting endstop status
Recv: x_min: open
Recv: y_min: open
Recv: z_min: TRIGGERED
Recv: z_probe: TRIGGERED

```
Looks OK, right?
But what if i deploy probe?
```
Send: M280 P0 S10
Recv: ok
[...]
Send: M119
Recv: Reporting endstop status
Recv: x_min: open
Recv: y_min: open
Recv: z_min: TRIGGERED
Recv: z_probe: open
```

Probe physically deploys, z_probe shows as "open", but z_min is shown as "TRIGGERED".
I can invert it's status (currently Z_MIN_ENDSTOP_INVERTING false) but the fact that it does not change it's state depending on probe status (just to remind - i have USE_PROBE_FOR_Z_HOMING set) - probably won't fix it.
There are no other physical switches on Z axis (only probe), why it is shown as "triggered" (hanging freely in air, could be raised or lowered without problems)?

**Comment by ABLomas** [comment]
Turns out that "#define USE_ZMIN_PLUG" was a problem.
I'm attaching my sort-of-working configuration (at least it homes and does not try to drill buildplate) in case someone needs.
[configuration_working.zip](https://github.com/MarlinFirmware/Marlin/files/4840812/configuration_working.zip)
